quantum/process_keycode/process_auto_shift.c

```diff
@@ -215,9 +219,31 @@ bool process_auto_shift(uint16_t keycode, keyrecord_t *record) {
                 return true;
 #    endif
         }
+#    ifdef RETRO_SHIFT
+        if (IS_RETRO(keycode) && (record->tap.count == 0 || record->tap.interrupted && (IS_LT(keycode))
+#        ifdef HOLD_ON_OTHER_KEYPRESS
```

Comment:
This is horrible but for some reason I can't think of a better way to do it. Send help.

Reply:
It's gotten worse but due to the `_per_key` stuff it definitely can't get better unfortunately.

Reply:
That one is nothing compared to `action_tapping.c`. . . the main if now takes up my entire screen.